sdk/horizon/streamer.go

```diff
@@ -60,12 +61,21 @@ func (h *Streamer) streamTx(cursor string, txs chan<- agent.StreamedTransaction,
 			Cursor: cursor,
 		}
 		err := h.HorizonClient.StreamTransactions(ctx, req, func(tx horizon.Transaction) {
+			txOrderedID, err := strconv.ParseInt(tx.PagingToken(), 10, 64)
+			if err != nil {
+				ctxCancel()
+				if h.ErrorHandler != nil {
+					h.ErrorHandler(err)
+				}
+				return
+			}
 			cursor = tx.PagingToken()
```

Comment:
💡  so it's clear these values are related/coming from the same source
```suggestion
			cursor = tx.PagingToken()
			txOrderedID, err := strconv.ParseInt(cursor, 10, 64)
			if err != nil {
				ctxCancel()
				if h.ErrorHandler != nil {
					h.ErrorHandler(err)
				}
				return
			}
```

Reply:
❓ for my own understanding, what's the difference between the the `cursor` and the `txOrderedID` ? ie. why can't we just save the cursor on the `escrowAccount.lastSeenTransactionOrderID`?

Reply:
They just happen to come from the same field in Horizon's response, but they are conceptually completely different things. The cursor is the identifier used to resume paging. The transaction order ID is an identifier that is unique for each transaction and guarantee to order transactions as they are processed on network.

I made a change not quite what you had in 410beb6, but similar.